process_report/tests/unit_tests.py

```diff
@@ -932,6 +932,410 @@ def test_two_pi(self):
+
+        self._assert_result_invoice(
+            test_invoice,
+            test_prepay_credits,
+            self.test_prepay_debits_file.name,
+            test_prepay_projects,
+            test_prepay_contacts,
+            answer_invoice,
+            answer_prepay_debits,
+            invoice_month,
+        )
+
+        # PI balance != Balance
+        test_invoice = self._get_test_invoice(project_names, [1000, 2000], [2000, 2500])
+
+        answer_invoice["Balance"] = [1000, 2000]
+
+        self._assert_result_invoice(
+            test_invoice,
+            test_prepay_credits,
+            self.test_prepay_debits_file.name,
+            test_prepay_projects,
+            test_prepay_contacts,
+            answer_invoice,
+            answer_prepay_debits,
+            invoice_month,
+        )
+
+    def test_two_group_one_project(self):
+        """How is prepay handled for two different groups with different credits and debits?"""
+        # Invoice month is before any credits are given
+        project_names = ["G1P1", "G2P1"]
+
+        invoice_month = "2024-03"
+        test_invoice = self._get_test_invoice(project_names, [1000, 2000])
+        test_prepay_credits = self._get_test_prepay_credits(
+            ["2024-04", "2024-04", "2024-06", "2024-08", "2024-10"],
+            ["G1", "G2", "G1", "G2", "G1"],
+            [700, 800, 1000, 2000, 3500],
+        )
+        test_prepay_debits = self._get_test_prepay_debits(
+            ["2024-05", "2024-06", "2024-07", "2024-10"],
+            ["G1", "G2", "G2", "G1"],
+            [200, 300, 1000, 2000],
+        )
+        test_prepay_debits.to_csv(self.test_prepay_debits_file.name, index=False)
+        test_prepay_projects = self._get_test_prepay_projects(
+            ["G1", "G2"], project_names, ["2024-01", "2024-01"], ["2024-12", "2024-12"]
+        )
+        test_prepay_contacts = self._get_test_prepay_contacts(
+            ["G1", "G2"], ["G1@bu.edu", "G2@harvard.edu"], [True, False]
+        )
+
+        answer_invoice = test_invoice.copy()
+        answer_invoice["Prepaid Group Name"] = ["G1", "G2"]
+        answer_invoice["Prepaid Group Institution"] = [
+            "Boston University",
+            "Harvard University",
+        ]
+        answer_invoice["MGHPCC Managed"] = [True, False]
+        answer_invoice["Prepaid Group Balance"] = [0, 0]
+        answer_invoice["Prepaid Group Used"] = [None, None]
+        answer_invoice["Invoice Email"] = ["G1@bu.edu", "G2@harvard.edu"]
+
+        answer_prepay_debits = test_prepay_debits.copy()
+
+        self._assert_result_invoice(
+            test_invoice.copy(),
+            test_prepay_credits,
+            self.test_prepay_debits_file.name,
+            test_prepay_projects,
+            test_prepay_contacts,
+            answer_invoice,
+            answer_prepay_debits,
+            invoice_month,
+        )
+
+        # Invoice month is after some credits and debits are given, TODO: Waiting for Kim's response
```

Comment:
Did you get the response you needed? Please make sure there are no TODOs left over in the PR.

Reply:
If not please link to what the question was.

Reply:
I believe my question was whether we would allow "backprocessing". You responded no. This was relevant to how I design the test case, and if we're not doing backprocessing, that will make things a lot simpler :)